packages/core/strapi/lib/services/entity-service/types/params/populate.d.ts

```diff
@@ -0,0 +1,45 @@
+import type { Attribute, Common, Utils } from '@strapi/strapi';
+
+/**
+ * Wildcard notation for populate
+ *
+ * To populate all the root level relations
+ */
+type WildcardNotation = '*';
+
+type PopulatableKeys<TSchemaUID extends Common.UID.Schema> = Utils.Guard.Never<
+  Attribute.GetPopulatableKeys<TSchemaUID>,
+  string
+>;
+
+/**
+ * Union of all possible string representation for populate
+ *
+ * @example
+ * type A = 'image'; // ✅
+ * type B = 'image,component'; // ✅
+ * type c = '*'; // ✅
+ * type D = 'populatableField'; // ✅
+ * type E = '<random_string>'; // ❌
+ */
+type StringNotation<TSchemaUID extends Common.UID.Schema> =
+  | WildcardNotation
+  | PopulatableKeys<TSchemaUID>
+  | `${string},${string}`
+  | `${PopulatableKeys}.${string}`;
+
+/**
+ * Array notation for populate
+ *
+ * @example
+ * type A = ['image']; // ✅
+ * type B = ['image', 'component']; // ✅
+ * type C = ['populatableField']; // ✅
+ * type D = ['<random_string>']; // ❌
+ */
+type ArrayNotation<TSchemaUID extends Common.UID.Schema> = StringNotation<TSchemaUID>[];
+
+export type Any<TSchemaUID extends Common.UID.Schema> =
+  | StringNotation<TSchemaUID>
```

Comment:
This may be an issue with documentation rather than this implementation, but the documentation for both [rest API](https://docs.strapi.io/dev-docs/api/rest/populate-select#population) and [entity service](https://docs.strapi.io/dev-docs/api/entity-service/populate) only allow `WildCardNotation | ArrayNotation | ObjectNotation` and do not mention strings as an option anywhere.

Reply:
Ok, so just a lack of documentation on those pages. I knew that the code supported it, it was a huge pain with the sanitization, I was curious if we could potentially drop any parts of it 😆

Pinging @pwizla in case we want to add simple attribute name strings and compound strings (like `"attribute.attribute"`) to the main documentation instead of only appearing suddenly in an advanced example.